src/Qowaiv.Validation.DataAnnotations/README.md

```diff
@@ -1 +1,128 @@
-﻿# Qowaiv Validation
\ No newline at end of file
+﻿# Qowaiv DataAnnotations based validation
+Provides an data annotations based implementation of the `Qowaiv.Validation.Abstractions.IValidator`
+and data annotation attributes.
+
+## Annotated model validator
+The `AnnotatedModelValidator` validates a model using its data annotations.
+It returns a `Result<TModel>` of the validated model:
+
+``` C#
+var validator = new AnnotatedModelValidator<SomeModel>();
+Result<SomeModel> result = validator.Validate(model);
+```
+
+## Validation messages
+`Qowaiv.Validation.DataAnnotations.ValidationMessage` inherits Microsoft's
+`ValidationResult`, and implements `Qowaiv.Validation.Abstractions.IValidationMessage`.
+This allows the creation of messages with different severities:
+
+``` C#
+var none = ValidationMessage.None;
+var info = ValidationMessage.Info(message, args);
+var warn = ValidationMessage.Warning(message, args);
+var error = ValidationMessage.Error(message, args);
+```
+
+## Validation attributes
+Multiple `[System.ComponentModel.DataAnnotations.Validation]` attributes can be
+used to decorate models.
+
+### Mandatory
+The `[Required]` attribute does not work for value types. The `[Mandatory]`
+attribute does. The default value of the `struct` is not valid. It also is not
+valid for the Unknown value, unless that is explicitly allowed.
+
+``` C#
+public class Model
+{
+    [Mandatory(AllowUnknownValue = true)]
+    public EmailAddress Email { get; set; }
+
+    [Mandatory()]
+    public string SomeString { get; set; }
+}
+```
+
+### Any
+The `[Required]` attribute does not work (well) for collections. The `[Any]`
+attribute does. It is only valid if the collection contains at least one item.
+
+``` C#
+public class Model
+{
+    [Any()]
+    public List<int> Numbers { get; set; }
+}
+```
+
+### Allowed values
+The `[AllowedValues]` attribute allows to define a subset of allowed values. It
+supports type converters to get the allowed values based on a string value.
+
+``` C#
+public class Model
+{
+    [AllowedValues("DE", "FR", "GB")]
+    public Country CountryOfBirth { get; set; }
+}
+```
+
+### Forbidden values
+The `[ForbiddenValues]` attribute allows to define a subset of forbidden values. It
+supports type converters to get the allowed values based on a string value.
```

Comment:
This line still says allowed values instead of forbidden values

Reply:
Ah, that was what you ment. :)